demos/node-classification/ppnp/appnp-reddit.ipynb

```diff
@@ -0,0 +1,415 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_dir = os.path.expanduser(\"~/data/reddit\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_reddit(data_dir):\n",
+    "\n",
+    "    with open(os.path.join(data_dir, \"reddit-G.json\")) as gfile:\n",
+    "        graph_data = json.load(gfile)\n",
+    "\n",
+    "    list_node_ids = list(d[\"id\"] for d in graph_data[\"nodes\"])\n",
+    "\n",
+    "    edge_generator = ((link[\"source\"], link[\"target\"]) for link in graph_data[\"links\"])\n",
+    "    edge_df = pd.DataFrame(edge_generator, columns=[\"target\", \"source\"])\n",
+    "\n",
+    "    with open(os.path.join(data_dir, \"reddit-class_map.json\")) as tfile:\n",
+    "        labels = json.load(tfile)\n",
+    "\n",
+    "    feats = np.load(data_dir + \"/reddit-feats.npy\")\n",
+    "\n",
+    "    feats[:, 0] = np.log(feats[:, 0] + 1.0)\n",
+    "    feats[:, 1] = np.log(feats[:, 1] - min(np.min(feats[:, 1]), -1))\n",
+    "\n",
+    "    feat_id_map = json.load(open(data_dir + \"/reddit-id_map.json\"))\n",
+    "\n",
+    "    # sort node features to match the order of feat_id_map\n",
+    "    sorted_idxs = np.array([feat_id_map[key] for key in list_node_ids])\n",
+    "    feats = feats[sorted_idxs, :]\n",
+    "    node_data = pd.DataFrame(feats)\n",
+    "\n",
+    "    # sort node labnels to match the order of feat_id_map\n",
+    "    labels = np.array([labels[key] for key in list_node_ids])\n",
+    "\n",
+    "    return edge_df, node_data, labels, list_node_ids, graph_data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of nodes: 231443\n",
+      "Number of edges: 11606919\n"
+     ]
+    }
+   ],
+   "source": [
+    "edge_df, node_data, labels, list_node_ids, graph_data = load_reddit(data_dir)\n",
+    "print(\"Number of nodes:\", len(node_data))\n",
+    "print(\"Number of edges:\", len(edge_df))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "target_encoding = preprocessing.OneHotEncoder(sparse=False, categories=\"auto\")\n",
+    "targets = target_encoding.fit_transform(labels.reshape(-1, 1))\n",
+    "targets = pd.DataFrame(targets)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then split the data into train/val/test based on the labels stored in the dataset. This is the same train/val/test split used in the graphsage paper. Then we fit a standard scaler on only the training data and use it to standardize all of the data."
```

Comment:
Maybe add the link to the GraphSAGE paper. 
Also, is this the same split as in the description above? Just checking. 
Talking about this from the description at the top 

> We use the first 20 days for training and the remaining days for testing (with 30% used for validation).

Reply:
Good idea re the link :+1:

Yeah I'm pretty sure its the same split. I got the graph off of snap and the nodes were labelled as train, test, and val and I use those labels to split the data. So i'm guessing they're the same splits used in the paper